### PR TITLE
Zachmu/engine test dolt

### PR DESCRIPTION
**Pull request by zachmu**
Modified test harness and data to work with doltdb

### CODE REVIEW

File: enginetest/enginetests.go [hunk 1]
```diff
@@ -185,33 +197,29 @@ func TestReadOnly(t *testing.T, harness Harness) {
...
 	})
+	require.NoError(t, err)
 
-	InsertRows(
-		t, mustInsertableTable(t,table),
-		sql.NewRow(int64(1), []interface{}{"a", "b"}, "first"),
-		sql.NewRow(int64(2), []interface{}{"c", "d"}, "second"),
-		sql.NewRow(int64(3), []interface{}{"e", "f"}, "third"),
-	)
+	InsertRows(t, harness.NewContext(), mustInsertableTable(t, table), sql.NewRow(int64(1), []interface{}{"a", "b"}, "first"), sql.NewRow(int64(2), []interface{}{"c", "d"}, "second"), sql.NewRow(int64(3), []interface{}{"e", "f"}, "third"), )
```
Comment (Hydrocharged) [inline enginetest/enginetests.go:215]: I prefer the old list format instead but it's nbd

File: sql/core.go [hunk 2]
```diff
@@ -260,6 +263,7 @@ type IndexAlterableTable interface {
 
 // InsertableTable is a table that can process insertion of new rows.
 type InsertableTable interface {
+	Table
```
Comment (Hydrocharged) [inline sql/core.go:266]: Seems a bit weird having `InsertableTable` implement `Table` but the other `-ableTable`s do not, but I guess it's also nbd

File: enginetest/enginetests.go (hunk 1 above)
Comment (zachmu) [reply enginetest/enginetests.go:215]: yeah me too, my editor did that and I was too lazy to undo it

File: sql/core.go (hunk 2 above)
Comment (zachmu) [reply sql/core.go:266]: I'll make this consistent. I did this for a dumb reason (debugging support), not attached to it.

**Pull request merged**